datastore_entity/src/lib.rs

```diff
@@ -141,6 +150,87 @@ where
     }
 }
 
+fn generate_empty_key(kind: String) -> Key {
+    Key {
+        partition_id: None,
+        path: Some(vec![PathElement {
+            id: None,
+            kind: Some(kind),
+            name: None,
+        }]),
+    }
+}
+
+pub fn commit_one<A>(
+    entity: DatastoreEntity,
+    kind: String,
+    token: A,
+    project_name: &String,
+) -> Result<DatastoreEntity, String>
```

Comment:
maybe possible to refactor later that `commit_one` is a special case of `commit_many` with some shared code...

Reply:
yes, I think most code can be reused, some thing for get_many/get_one